plugins/modules/cloud/misc/terraform.py

```diff
@@ -345,6 +350,7 @@ def main():
             lock=dict(type='bool', default=True),
             lock_timeout=dict(type='int',),
             force_init=dict(type='bool', default=False),
+            overwrite_init=dict(type='bool', default=True),
```

Comment:
Why should this be enabled by default?

This is going to break someone's workflow :P https://xkcd.com/1172/

Reply:
Because otherwise it would be a breaking change. The default `True` is the only way to make the condition in https://github.com/ansible-collections/community.general/pull/2573/files#diff-ae011ea8a32066685f6596e6c8cc79928698d52715cb716a5151e5bff4658f8dR393 always evaluate to `True`.

Reply:
> When a Terraform workspace is already _init_, with a foreign backend creds (by S3 creds from another workspace), Terraform fails to redo _init_.

I'm not sure if someone has been relying on this error or not (fail on already initialized workspace).
The thing I'm saying sounds pretty much similar to https://xkcd.com/1172/ really :)

Reply:
The code with this PR still does exactly the same as before this PR. Only when the user explicitly passes `overwrite_init: false`, then the behavior changes (by not retrying an init when that tfstate file is already there).

Reply:
Sorry, now I realized what I was missing, thanks for clarification!

Reply:
No problem, it's better to discuss about such things too much instead of missing something important :)

Reply:
> The code with this PR still does exactly the same as before this PR. Only when the user explicitly passes `overwrite_init: false`, then the behavior changes (by not retrying an init when that tfstate file is already there).

Yes, this is exactly that :)